client/src/components/__tests__/ElicitationModal.test.tsx

```diff
@@ -0,0 +1,535 @@
+          });
+        });
+
+        await act(async () => {
+          fireEvent.click(screen.getByTestId("button-submit"));
+        });
+
+        expect(mockResolve).toHaveBeenCalledWith({
+          action: "accept",
+          content: { name: "John", email },
+        });
+
+        jest.clearAllMocks();
+        document.body.innerHTML = "";
+      }
+    });
+
+    it("should reject invalid email formats", async () => {
+      const invalidEmails = [
+        "invalid-email",
+        "@example.com",
+        "user@",
+        "user.example.com",
+      ];
+
+      for (const email of invalidEmails) {
+        renderElicitationModal();
+
+        const input = screen.getByTestId("form-input");
+        await act(async () => {
+          fireEvent.change(input, {
+            target: { value: `{"name": "John", "email": "${email}"}` },
+          });
+        });
+
+        await act(async () => {
+          fireEvent.click(screen.getByTestId("button-submit"));
+        });
+
+        expect(
+          screen.getByText(/Invalid email format: email/),
+        ).toBeInTheDocument();
+        expect(mockResolve).not.toHaveBeenCalledWith(
+          expect.objectContaining({ action: "accept" }),
+        );
+
+        jest.clearAllMocks();
+        document.body.innerHTML = "";
+      }
+    });
+  });
+
+  describe("Schema Types", () => {
+    it("should handle string schema", () => {
+      const request = createMockRequest({
+        requestedSchema: {
+          type: "object",
+          properties: {
+            username: { type: "string", description: "Username" },
+          },
+        },
+      });
+      renderElicitationModal(request);
+      expect(screen.getByTestId("dynamic-json-form")).toBeInTheDocument();
+    });
+
+    it("should handle number schema", () => {
+      const request = createMockRequest({
+        requestedSchema: {
+          type: "object",
+          properties: {
+            age: { type: "number", minimum: 0, maximum: 120 },
+          },
+        },
+      });
+      renderElicitationModal(request);
+      expect(screen.getByTestId("dynamic-json-form")).toBeInTheDocument();
+    });
+
+    it("should handle boolean schema", () => {
+      const request = createMockRequest({
+        requestedSchema: {
+          type: "object",
+          properties: {
+            subscribe: {
+              type: "boolean",
+              description: "Subscribe to newsletter",
+            },
+          },
+        },
+      });
+      renderElicitationModal(request);
+      expect(screen.getByTestId("dynamic-json-form")).toBeInTheDocument();
+    });
+
+    it("should handle enum schema", () => {
+      const request = createMockRequest({
+        requestedSchema: {
+          type: "object",
+          properties: {
+            country: {
+              type: "string",
+              enum: ["US", "UK", "CA"],
+              enumNames: ["United States", "United Kingdom", "Canada"],
```

Comment:
`enumNames` [is not part of the JSONSchema spec](https://github.com/rjsf-team/react-jsonschema-form/issues/532). The [recommended way to describe](https://github.com/json-schema-org/json-schema-spec/issues/57) enum labels is with `oneOf` or `anyOf`.